BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -505,7 +505,12 @@ GatherDeviceInfo (
     }
   }
 
-  PciExpressWrite16 (PCI_EXPRESS_LIB_ADDRESS (Bus, Device, Func, PCI_COMMAND_OFFSET), Value);
+  //
+  // Restore PCI command register only if the original bus master was enabled.
+  //
+  if ((Value & EFI_PCI_COMMAND_BUS_MASTER) != 0) {

```

Comment:
Will this disable UART debug msg ?  I guess it might be good to keep this ?

Reply:
This will not disable UART debug message since UART PCI device already enabled its bus master before PCI enumeration.

Reply:
UART does not have to enable bus master.  For example, it can just enable MMIO or IO alone.  In that case, this will disable the UART, right ?

Reply:
Okay. Got it what you mean. Let me remove this unnecessary if condition.